lightning/src/offers/invoice.rs

```diff
@@ -967,6 +976,13 @@ impl Bolt12Invoice {
 		self.tagged_hash.as_digest().as_ref().clone()
 	}
 
+	/// Returns the [`OfferId`] if this invoice corresponds to an [`Offer`].
+	///
+	/// [`Offer`]: crate::offers::offer::Offer
+	pub fn offer_id(&self) -> Option<OfferId> {
+		self.offer_id
+	}
```

Comment:
Ah, sorry... we should put this in `invoice_accessors_common` such that `UnsignedBolt12Invoice`, `UnsignedStaticInvoice`, and `StaticInvoice` also get this method.